src/main/resources/static/js/courseform.js

```diff
@@ -0,0 +1,73 @@
+$(document).ready(function() {
+    $('#reuse-grid-button').on('click', function (e) {
+        //Display Course Search Modal
+        $('#search-course-grid-modal')
+            .css('display', 'block')
+            .css('opacity', '1');
+
+        //Cancel, Close Modal, Restore Previous State
+        $('#search-course-grid-modal-close-top')
+            .add('#search-course-grid-modal-close-bottom')
+            .on('click', function (e) {
+                $('#search-course-grid-modal')
+                    .css('display', 'none')
+                    .css('opacity', '0');
+
+                $('#course-grid-id').attr('value', '');
+                $('#search-course-field').val('');
+                $('.course-list-queried-item').css('background-color', 'initial');
+            });
+
+        //Close Modal After Submit
+        $('#search-course-grid-modal-submit-bottom').on('click', function(e) {
+            $('#search-course-grid-modal')
+                .css('display', 'none')
+                .css('opacity', '0');
```

Comment:
I'm seeing a recurring thene here. Might be worth to refactor a function? Modularity will be especially nice if this modal stuff is going to be reused (which I would like to for my issue if possible, to keep style and feel consistent)

Reply:
Agreed. Ill see what I can do